File: frontend/src/types/certificates.ts
export interface Certificate {
  id: string
  course_id: string | null
  learning_path_id: string | null
  title: string
  certificate_number: string
  issued_at: string
}

export interface CertificateVerification {
  certificate_number: string
  recipient_name: string
  course_title: string
  issued_at: string
}

File: backend/app/schemas/certificates.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CertificateRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    course_id: uuid.UUID | None
    learning_path_id: uuid.UUID | None
    title: str
    certificate_number: str
    issued_at: datetime


class CertificateVerification(BaseModel):
    certificate_number: str
    recipient_name: str
    course_title: str
    issued_at: datetime
